### PR TITLE
feat(web-modeler): add git-sync self-hosted part

### PR DESCRIPTION
**Pull request by JacobiCamunda**
## Description

Add documentation part when using self-hosted GitHub instances (See [PR](https://github.com/camunda/web-modeler/pull/11809)). We will release a release-candidate for 8.6 today so that a customer can already run some tests, and it should be released somewhere today or tomorrow.

## When should this change go live?

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**.
- [ ] This is already available but undocumented and should be released within a week.
- [ ] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR)
- [ ] This is part of a scheduled alpha or minor. (apply alpha or minor label)
- [ ] There is **no urgency** with this change and can be released at any time.
- [x] This will be part of a release-candidate (released today 20.11.24)

## PR Checklist

- [x] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [x] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineering DRI or delegate.
- [ ] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @camunda/tech-writers as a reviewer.

### CODE REVIEW

**Review by marcellobarile — APPROVED** [review]

File: docs/components/modeler/web-modeler/git-sync.md
Comment (jfriedenstab) [reply docs/components/modeler/web-modeler/git-sync.md:46]: Probably, it would also make sense to document this on https://docs.camunda.io/docs/self-managed/modeler/web-modeler/configuration/#configuration-of-the-restapi-component, but this could also be done later (together with the GitLab base URL).

**Review by jfriedenstab — CHANGES_REQUESTED** [review]

**Review by jfriedenstab — DISMISSED** [review]
👍🏻

File: docs/components/modeler/web-modeler/git-sync.md
Comment (conceptualshark) [inline docs/components/modeler/web-modeler/git-sync.md:48]: I'm wondering if this might be more apparent to a reader if included as part of the steps? This may make it more visible vs two back-to-back `:::note` blocks. For example:

```suggestion
1. To use a **self-hosted GitHub instance,** set the environment variable `CAMUNDA_MODELER_GITSYNC_GITHUB_BASEURL` to the API URL of your self-hosted GitHub instance (for example, `http(s)://HOSTNAME/api/v3`). Refer to the [GitHub documentation](https://docs.github.com/en/enterprise-server@3.15/rest/enterprise-admin?apiVersion=2022-11-28#endpoint-urls) and choose the correct Enterprise Server version.
```

Also a small wording suggestion.

File: docs/components/modeler/web-modeler/git-sync.md
Comment (conceptualshark) [inline docs/components/modeler/web-modeler/git-sync.md:137]:
```suggestion
- Using self-hosted instances of Git providers may require additional configuration. Refer to the Web Modeler [configuration](#configure-github-in-web-modeler) for more details.
```

File: docs/components/modeler/web-modeler/git-sync.md
Comment (JacobiCamunda) [reply docs/components/modeler/web-modeler/git-sync.md:48]: Mhm, good point, but I don't know if I prefer it. @marcellobarile, any opinion here as the DRI?

Comment (marcellobarile) [reply docs/components/modeler/web-modeler/git-sync.md:48]: > I'm wondering if this might be more apparent to a reader if included as part of the steps

To be noted that it's not a mandatory step. Could this better drive the decision?

Comment (JacobiCamunda) [reply docs/components/modeler/web-modeler/git-sync.md:48]: I will leave it as is for now and revisit later when GitLab is added.

File: docs/components/modeler/web-modeler/git-sync.md
Comment (conceptualshark) [inline docs/components/modeler/web-modeler/git-sync.md:54]:
```suggestion
   - **Client ID:** Found in your GitHub App's settings page. You can also use Application ID as an alternative. (If you are using GitHub Enterprise Server 3.13 or prior, Application ID is required.)
```

**Review by conceptualshark — DISMISSED** [review]
A small suggestion, and I do think some of the wording changes suggested for the note could be helpful, but both are non-blocking and I am good with where this PR is!

**Comment by JacobiCamunda** [comment]
Could someone approve this again so it can be merged, @jfriedenstab or @conceptualshark? I forgot about this because I enabled auto-merge, but it wasn't auto-merged because of a missing approval.

**Review by mesellings — APPROVED** [review]
Not checked content, approving to unblock as per request from @JacobiCamunda 👍

**Review by jfriedenstab — APPROVED** [review]

**Pull request merged**
